src/lib/util/string_array.c

```diff
@@ -98,14 +98,20 @@ string_array_has_value(char **array, const char *value)
     }
 
     for (i = 0; array[i] != NULL; i++) {
-        if (strcmp(value, array[i]) == 0) {
+        if (strncmp(value, array[i], len) == 0) {
```

Comment:
I think that this will fail when there is longer value with the same beginning in the array. Imagine
* have in array "hello"
* add "hell", 4

Reply:
Fixed. Good catch.